cmd/agent/api/agent/agent.go

```diff
@@ -251,6 +251,8 @@ func streamLogs(w http.ResponseWriter, r *http.Request) {
 			// The buffer will flush on its own most of the time, but when we run out of logs flush so the client is up to date.
 			flusher.Flush()
 		}
+		// don't hog the CPU
+		time.Sleep(1 * time.Millisecond)
```

Comment:
`logMessageReceiver.Next` is implemented as
```
for {
		select {
		case msgPair := <-b.inputChan:
			if shouldHandleMessage(msgPair.msg, filters) {
				return formatMessage(msgPair.msg, msgPair.redactedMsg), true
			}
			continue
		default:
			return "", false
		}
	}
```

What do you think about updating `logMessageReceiver.Next` to push the log line in a channel instead of returning the log. This channel could be listened in the [select](https://github.com/DataDog/datadog-agent/pull/8080/files#diff-69cfabc8839fba7cdf69e2f00e951d679f86281269bfd8a6f0a3dd3666964bf0R241).
You could remove the [default](https://github.com/DataDog/datadog-agent/pull/8080/files#diff-69cfabc8839fba7cdf69e2f00e951d679f86281269bfd8a6f0a3dd3666964bf0R246) and It should avoid the busy loop.
Note: `flusher.Flush()` may be called at a regular interval with a timer.

Reply:
That's indeed a nice design improvement. Creating a follow-up card for that, while moving forward with this fix as a temporary solution if that's ok.